packages/contracts/src/ReputationProof.ts

```diff
@@ -0,0 +1,83 @@
+import { BigNumberish, utils } from 'ethers'
+import { SnarkProof } from '@unirep/crypto'
+import UnirepCircuit, { CircuitConfig, CircuitName } from '@unirep/circuits'
+
+import { UnirepTypes } from './contracts/IUnirep'
+import path from 'path'
+
+export class ReputationProof implements UnirepTypes.ReputationProofStruct {
+    public repNullifiers: BigNumberish[]
+    public epoch: BigNumberish
+    public epochKey: BigNumberish
+    public globalStateTree: BigNumberish
+    public attesterId: BigNumberish
+    public proveReputationAmount: BigNumberish
+    public minRep: BigNumberish
+    public proveGraffiti: BigNumberish
+    public graffitiPreImage: BigNumberish
+    public proof: BigNumberish[]
+    private publicSignals: BigNumberish[]
+    private zkFilesPath: string
+    private config: CircuitConfig
+
+    constructor(
+        _publicSignals: BigNumberish[],
+        _proof: SnarkProof,
+        _zkFilesPath: string
+    ) {
+        this.zkFilesPath = _zkFilesPath
+        this.config = require(path.join(this.zkFilesPath, 'config.json'))
+        const formattedProof: any[] =
+            UnirepCircuit.formatProofForVerifierContract(_proof)
+        this.repNullifiers = _publicSignals.slice(
+            0,
+            this.config.maxReputationBudget
+        )
+        this.epoch = _publicSignals[this.config.maxReputationBudget]
+        this.epochKey = _publicSignals[this.config.maxReputationBudget + 1]
+        this.globalStateTree =
+            _publicSignals[this.config.maxReputationBudget + 2]
+        this.attesterId = _publicSignals[this.config.maxReputationBudget + 3]
+        this.proveReputationAmount =
+            _publicSignals[this.config.maxReputationBudget + 4]
+        this.minRep = _publicSignals[this.config.maxReputationBudget + 5]
+        this.proveGraffiti = _publicSignals[this.config.maxReputationBudget + 6]
+        this.graffitiPreImage =
+            _publicSignals[this.config.maxReputationBudget + 7]
+        this.proof = formattedProof
+        this.publicSignals = _publicSignals
+    }
+
+    public verify = (): Promise<boolean> => {
+        const proof_ = UnirepCircuit.formatProofForSnarkjsVerification(
+            this.proof.map((n) => n.toString())
+        )
+        return UnirepCircuit.verifyProof(
```

Comment:
I'm not a fan of this structure. The dependency structure here looks like this:

`app` -> `@unirep/contracts` -> `@unirep/circuits` -> `snarkjs`

I think it's reasonable to to a more minimalist approach like this:

`app` -> `snarkjs`

It seems like wrapping `snarkjs` with a helper function (like this) makes things more simple, but it actually makes things harder because developers have to learn a wrapper function name/args for every project (e.g. unirep/interep/zkopru/semaphore). Then if there's a bug the developer has to figure out if it's the app, the unirep wrapper logic, or snarkjs. It's much easier to just get a proof and give it to snarkjs IMO.